k2/csrc/top_sort_test.cu

```diff
@@ -198,14 +198,17 @@ TEST(TopSort, RandomVectorOfFsas) {
 // another random test which uses IsRandEquivalent to check the result
 TEST(TopSort, RandomVectorOfFsas1) {
   ContextPtr cpu = GetCpuContext();
-  for (int32_t n = 0; n != 0; ++n) {
-    // for (auto &context : {GetCpuContext(), GetCudaContext()}) {
-    for (auto &context : {GetCpuContext()}) {
+  for (int32_t n = 0; n != 1; ++n) {
+    for (auto &context : {GetCpuContext(), GetCudaContext()}) {
       K2_LOG(INFO) << n;
       // We have to generate epsilon-free Fsas here as we will call Intersect
       // below (in `IsRandEquivalent`) which requires at least one of the two
       // input Fsas is epsilon-free.
-      FsaVec random_fsas = RandomFsaVec(1, 1000, false, true);
+      // TODO(haowen): here we generate acyclic Fsas for testing as current
+      // implementation of GPU TopSort supposes the input Fsas are acyclic, but
+      // definitely I need to `reorder` states randomly and add some self-loops
+      // for robust test.
+      FsaVec random_fsas = RandomFsaVec(1, 1000, true, true);
```

Comment:
We could use RenumberFsa to do this, perhaps?  Would be easiest to generate the reordering randomly on CPU (not touching start+final states, of course).

Reply:
yes, that's exactly what I was planning to do, but not in this PR.